### ISSUE TITLE
When Rack is re-opened after a crash, the patch filename is sometimes not updated

### ISSUE THREAD

**Issue by XtremeSounds** [VCVRack/Rack#1749]
windows 10 

I loaded an old patch template and started creating.  I then applied the cf  Label module and when I went to rename it vcv crashed. When I reloaded vcv up again it pulled up the last instance of what I was working before the crash however it has a default patch name as a different patch I had been working on a few days ago.  I then hit save to protect it without realizing the old name and vcv overwrote the other patch I was working on before I made this one. So I lost my other patch entirely and the labels module is crashing vcv.  Strange behavior

**Comment by AndrewBelt** [comment]
I can't reproduce this. So the steps to reproduce is the following?
1. Have patch A open. Rack's patch filename is `A.vcv`.
2. Open patch B. Rack's patch filename is `B.vcv`.
3. Rack crashes.
4. Reopen Rack.
5. Patch B is on the screen, but Rack's patch filename is `A.vcv`.

The only way I can think of this happening is if during shutdown, patch autosaving succeeds but `settings.json` saving crashes. I can't think of a reason that settings saving would ever crash.

**Comment by modman2019** [comment]
I think what happened was i loaded vcv rack which opened up patch B which was the last one I was working on. Then I decided to open to open up patch C which was my mixer template patch. i then created patch A from patch C  and hit save and named it. Then I decided to add the label module which crashed patch A. When vcv reloaded it opened up patch Bs name but displayed patch As connected before the label and I hjit save without checking the name assuming it would display patch As name.  Patch B is now a duplicate of patch A,

I wish VCV had a sandbox mode which prevented crashes when plugins fail.

**Comment by AndrewBelt** [comment]
>created patch A from patch C

Can you be more clear on this? Do you mean you used "File > Save as"?

>I wish VCV had a sandbox mode which prevented crashes when plugins fail.

You design it, I'll implement it. (Hint: It's impossible)

**Comment by modman2019** [comment]
yes i used save as and renamed it

**Comment by EmergencyPeanut** [comment]
I think this a duplicate of #1559 I get this all the time. I assume  autosaving is not happening as rack has crashed.

Worse today for the first time it didn't  revert  after crashing to the last saved name rather than the current name but to a name I have not used in months. A fix would be most welcome.

**Comment by gabrielreygoodlatte** [comment]
This happens to me regularly too. After a crash (usually caused by trying to switch audio drivers), I reopen VCV and I seem to have lost no work. However, the file name is the name of the second-to-last file I was working on - that's consistent. I've usually noticed it, and intentionally done a save-as to avoid overwriting anything, but this time I hit ctrl-s and it overwrote another patch that I had just listened to. I don't know how autosave works, but it seems like whatever records the file's data, and whatever records the associated file name are out of sync. It's like the data always updates properly, but the filename only updates on a successful exit from VCV (not a crash), or something like that. If I manually exit VCV without saving, but there is no crash, this never happens. I'm also on Windows 10.

**Comment by nickfeisst** [comment]
This just happened to me. I opened Rack, which opened a patch that I've been working on for around 5+ hours over the last couple of days. I then created a new patch, and after some time Rack crashed. I reopened Rack and my new patch, which hadn't yet been saved by me, was still there, so I intuitively pushed "command-s" to make sure it saved. It was only after I wasn't prompted to select a name for the file that I realised that the name of the file was the same as my first patch.

This bug has me a little nervous now as all that work has been lost so easily.

[System: macOS 11.0.1 / X86-64 (Translated) on an Apple M1 CPU]

**Comment by AndrewBelt** [comment]
Duplicate of https://github.com/VCVRack/Rack/issues/1559, see that issue for latest update.

**Comment by vuzhmusic** [comment]
This happens to me as well, it's very upsetting.  I've lost several great patches this way.

Maybe a workaround would be to have patch files have ability to restore to prior save states.  Would increase the size of patch files, but probably not by too much.

**Comment by EmergencyPeanut** [comment]
> This happens to me as well, it's very upsetting. I've lost several great patches this way.
> 
> Maybe a workaround would be to have patch files have ability to restore to prior save states. Would increase the size of patch files, but probably not by too much.

Not my place to say but a workaround is not needed in this situation, it just needs fixing; it's one of those things that happens when an application shuts down unexpectantly.
